plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/IOpenShiftConnection.java

```diff
@@ -38,6 +38,8 @@
 	
 	<T extends IResource> List<T> getResources(String kind, String namespace);
 
+	<T extends IResource> T getResource(IResource resource);
```

Comment:
That feels pretty confusing from an API perspective. Get a resource from a resource. At the very least this requires some Javadoc